CMakeLists.txt

```diff
@@ -26,6 +26,16 @@ if(CXX_COMPILER_IS_DPCPP)
   message(STATUS "CXX compiler at ${CMAKE_CXX_COMPILER} detected as DPC++")
 endif()
 
+if ((CMAKE_CURRENT_BINARY_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR) OR
+  (CMAKE_CURRENT_BINARY_DIR STREQUAL ${CMAKE_CURRENT_SOURCE_DIR}/src) OR
```

Comment:
I don't think we should attempt to prevent *every* source directory from being used as a build directory. If a user really wants to build in `src`, I say let them...

Reply:
ok, allowing to build in sub dirs will be allowed, but I rather keep a custom error message, since the error message from the undocumented built-in vars is not as direct and clear as a custom one can be.